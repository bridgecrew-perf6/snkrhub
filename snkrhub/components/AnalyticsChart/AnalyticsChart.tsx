import React from 'react';
import { Dimensions, Appearance } from 'react-native';
import { VictoryArea, VictoryChart, VictoryClipContainer, VictoryAxis } from "victory-native";
import { FormattedAnalyticsData } from '../../screens/AppScreens/types';

// Components

// Types
type Props = {
  analyticsRangeData: FormattedAnalyticsData[] | undefined
}

const AnalyticsChart = (props: Props) => {
  return (
    <>
    {props.analyticsRangeData &&
      <VictoryChart 
        width={Dimensions.get("window").width}
        // containerComponent={
        //   <VictoryVoronoiContainer voronoiDimension="x"
        //     labels={({ datum }) => `USD$${datum.inventoryvalue}`}
        //     labelComponent={
        //       <VictoryTooltip cornerRadius={0} flyoutStyle={{fill: "#71717a"}}/>
        //     }
        //   />
        // }
        padding={{ top: 10, bottom: 80, left: 40, right: 50 }}
      >
        <VictoryArea 
          data={props.analyticsRangeData} 
          y="inventoryvalue" 
          interpolation="natural"
          groupComponent={
            <VictoryClipContainer 
              clipPadding={{ top: 5, right: 0 }}
            />
          }
          animate={{
            duration: 1000,
            onLoad: { duration: 1000 }
          }}
          style={{ 
            data: { 
              stroke: "#2563eb", 
              strokeWidth: 3.5, 
              strokeLinecap: "round" ,
              fill: 'rgba(37, 99, 235, 0.20)',

            } 
          }}
        />
        <VictoryAxis
          dependentAxis
          tickFormat={(y) => y}
          style={{
            tickLabels: {
              fill: '#71717a',
              fontSize: 12
            }, 
            grid: {
              stroke: 'rgb(143, 155, 179, 0.40)',
              strokeDasharray: '7',
            }
          }}
        />
      </VictoryChart>
      }
    </>
  )
}

export default AnalyticsChart