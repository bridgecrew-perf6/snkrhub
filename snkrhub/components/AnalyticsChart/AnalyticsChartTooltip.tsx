import { useEffect } from 'react';
import Svg, { Line, G, Text as TextSvg, Rect } from 'react-native-svg';

type Props = {
    position: { x: number, y: number},
    value: { x: number, y: number, meta?: any },
    changeInventoryValue: (amount: number) => void
}

const AnalyticsChartTooltip = (props?: any) => {
    const { x, y } = props.position;
    const { x: xValue, y: yValue } = props.value;

    useEffect(() => {
        props.changeInventoryValue(yValue)
    }, [yValue])

    return (
        <>
            <G>
                {/* Tooltip */}
                <G x={x - 63 / 2} y={-20}>
                    <Rect height={27} width={63} fill={'#27272a'} rx={4} />
                    <TextSvg x={63 / 2} dy={18} textAnchor={'middle'} stroke={'white'}>
                    {`${yValue}`}
                    </TextSvg>
                </G>

                {/* Ball */}
                <G x={x - 4} y={y - 5}>
                    <Rect
                    height={10}
                    width={10}
                    fill={'#2563eb'}
                    stroke={'#151a30'}
                    rx={100}
                    />
                </G>

                {/* Graph Line */}
                <Line
                    key={'zero-axis'}
                    x1={x + 2}
                    x2={x}
                    y1={7}
                    y2={160}
                    stroke={'rgb(143, 155, 179, 0.40)'}
                    // style={{ opacity: 0.3 }}
                    strokeWidth={1.5}
                />
            </G>
        </>
    )
}

export default AnalyticsChartTooltip