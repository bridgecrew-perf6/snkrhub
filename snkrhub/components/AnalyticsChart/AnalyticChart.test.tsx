import * as React from 'react';
import renderer from 'react-test-renderer';

import { FontAwesome5 } from '@expo/vector-icons';
import { Icon } from "native-base";
import AnalyticsChart from './AnalyticsChart';
import { NativeBaseProvider } from 'native-base';

// test that inventory item accepts correct props
it(`analytics chart correctly renders data`, () => {
  const props = {
    analyticsRangeData: [
        { inventoryvalue: 130 },
        { inventoryvalue: 165 },
        { inventoryvalue: 142 },
        { inventoryvalue: 190 }
    ]
  }

  const tree = renderer.create(
    <NativeBaseProvider>
      <AnalyticsChart {...props} />
    </NativeBaseProvider>
  ).toJSON()

  expect(tree).toMatchSnapshot();
});
