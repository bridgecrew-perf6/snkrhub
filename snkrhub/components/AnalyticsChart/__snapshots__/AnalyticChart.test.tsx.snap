// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analytics chart correctly renders data 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
